core/src/main/java/com/sequenceiq/cloudbreak/service/freeipa/FreeIpaCleanupService.java

```diff
@@ -83,6 +83,13 @@ public void cleanup(Stack stack, boolean hostOnly, boolean recover, Set<String>
     }
 
     private void pollCleanupOperation(OperationStatus operationStatus) {
+        // TODO Question: Is a cleanup operation required?
```

Comment:
I think this is necessary, --force does not override the DNS entries it just appends a new IP to the existing hostname causing all sorts of issues for Kerberos.

Reply:
yes, this is required

Reply:
Ack. Would still be interested in eventually making this faster. This seems to take longer with more nodes being added (not linear growth though)